rfcs/0000-buildpack-yml-to-env-var.md

```diff
@@ -0,0 +1,86 @@
+# Buildpack.yml to Environment Variables
+
+## Proposal
+
+Migrate to using environment variables to do all buildpack configuration and
+get rid of `buildpack.yml`.
+
+## Motivation
+
+There are several reasons for making this switch.
+1. There is already an existing RFC that proposes moving away from
+   `buildpack.yml` as a configuration tool.
+1. Environment variables appears to be the standard for configuration in other
+   buildpack ecosystems such as Google Buildpacks and Heroku as well as the
+   Paketo Java buildpacks. Making this change will align the buildpack with the
+   rest of the buildpack ecosystem.
+1. There is native support to pass environment variables to the buildpack
+   either on a per run basis or by configuration that can be checked into
+   source control, in the form of `project.toml`.
+
+## Implementation
+
+The proposed environment variables for Go Build are as follow:
+
+#### BP_GO_BUILD_FLAGS
+
+```shell
+$BP_GO_BUILD_FLAGS='-buildmode=default,-tags=paketo,-ldflags="-X main.variable=some-value"'
+```
+
+This will replace the following structure in `buildpack.yml`:
+
+```yaml
+go:
+  build:
+    flags:
+    - -buildmode=default
+    - -tags=paketo
+    - -ldflags="-X main.variable=some-value"
+```
+
+#### BP_GO_BUILD_IMPORT_PATH
+
+```shell
+$BP_GO_BUILD_IMPORT_PATH=example.com/some-app
+```
+
+This will replace the following structure in `buildpack.yml`:
+
+```yaml
+go:
+  build:
+    import-path: example.com/some-app
+```
+
+#### BP_GO_TARGETS
+
+```shell
+$BP_GO_TARGETS=./first:./second
```

Comment:
Given the example below, should this say:
```
$BP_GO_TARGETS=./cmd/web-server:./cmd/debug-server
```

Reply:
Yes that is the case. Let me update that!